swift/apple/FirezoneNetworkExtension/SystemConfigurationResolvers.swift

```diff
@@ -102,7 +102,13 @@ class SystemConfigurationResolvers {
     guard let result = SCDynamicStoreCopyValue(dynamicStore, path as CFString)
     else {
       let code = SCError()
-      Log.error(SystemConfigurationError.unableToCopyValue(path: path, code: code))
+
+      // kSCStatusNoKey indicates the key is missing, which is expected if the
+      // interface has no DNS configuration.
+      if code != kSCStatusNoKey {
+        Log.error(SystemConfigurationError.unableToCopyValue(path: path, code: code))
+      }
```

Comment:
I'd suggest refactoring this to early exit instead:

```
if code == kSCStatusNoKey {
	return nil
}

Log.error(SystemConfigurationError.unableToCopyValue(path: path, code: code))
```

Reply:
Hm yeah I can do that. Note though we need to explicitly return nil, so we'll have another `return nil` after the `Log.error` call.